Entities/AHuman.cpp

```diff
@@ -1810,20 +1809,20 @@ void AHuman::UpdateWalkAngle(AHuman::Layer whichLayer) {
 			rayLength += m_pBGLeg->GetMaxLength();
 			hipPos += RotateOffset(m_pBGLeg->GetParentOffset());
 		}
-		float traceRotation = c_HalfPI / static_cast<float>(rayCount - 1) * GetFlipFactor();
-		Vector hitPos;
-		Vector terrainVector(0, rayLength);
-		Vector trace(0, rayLength);
-		for (int i = 0; i < rayCount; i++) {
-			if (g_SceneMan.CastStrengthRay(hipPos, trace, 10.0F, hitPos, 4, g_MaterialGrass)) {
-				terrainVector += trace - g_SceneMan.ShortestDistance(hipPos, hitPos, g_SceneMan.SceneWrapsX());
-			} else {
-				// Since no terrain was found, reinforce the existing terrain direction Vector, so any terrain found in future affects it less.
-				terrainVector *= 1.5F;
-			}
-			trace.RadRotate(traceRotation);
-		}
-		m_WalkAngle[whichLayer] = Matrix((terrainVector * GetFlipFactor()).GetAbsRadAngle() + c_HalfPI * GetFlipFactor());
+
+		// Cast a ray down from the left and right of us, to determine our angle of ascent
+		Vector hitPosLeft = hipPos + Vector(-10.0F, 0.0F);
```

Comment:
I'm really not sure what the old code was trying to do. I think 4zK added and I probably looked at it and decided I didn't want to try and figure it out, cause it's confusing.

Anyway, the only issue I can see with your stuff is the hardcoded hitPos modifier of +-10x. Might have problems with different sized actors?

Reply:
I suppose a flat 10px isn’t ideal, ideally it’d be tied to stride length instead, but… meh?

Reply:
Yeah, the magic number works well enough for now. I'll chuck in a todo to change it sometime.
And if you wanna come up with something for it, feel free to do so of course.